packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -36,32 +53,57 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
     AppLifecycleProvider? appLifecycleProvider,
     DeviceContextInfoProvider? deviceContextInfoProvider,
     required Connect dbConnectFunction,
-  })  : _keyValueStore = keyValueStore,
+  })  : _keyValueStore = keyValueStore ??
+            AmplifySecureStorageWorker(
+              config: AmplifySecureStorageConfig(
+                scope: 'analyticsPinpoint',
+              ),
+            ),
         _pathProvider = pathProvider,
         _appLifecycleProvider = appLifecycleProvider,
         _deviceContextInfoProvider = deviceContextInfoProvider,
         _dbConnectFunction = dbConnectFunction;
 
-  AnalyticsClient? __analyticsClient;
-  AnalyticsClient get _analyticsClient {
-    if (__analyticsClient == null) {
+  void _ensureConfigured() {
+    if (!_isConfigured) {
       throw const AnalyticsException(
         'Analytics not configured',
         recoverySuggestion:
             'Please make sure to call: await Amplify.configure(amplifyconfig)',
       );
     }
-    return __analyticsClient!;
   }
 
-  final SecureStorageInterface? _keyValueStore;
+  var _isConfigured = false;
+  var _analyticsEnabled = false;
+
+  @visibleForTesting
+
+  /// Storage key for the static Pinpoint endpoint id
+  static const String endpointIdStorageKey = 'UniqueId';
```

Comment:
Auto formatting is adding the extra whitespace because the comment should be first.
```suggestion
  /// Storage key for the static Pinpoint endpoint id
  @visibleForTesting
  static const String endpointIdStorageKey = 'UniqueId';
```